pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go

```diff
@@ -598,9 +597,94 @@ func getSubnetIDForLB(compute *gophercloud.ServiceClient, node v1.Node) (string,
 	return "", ErrNotFound
 }
 
+// getPorts gets all the filtered ports.
+func getPorts(network *gophercloud.ServiceClient, deviceID string, tagsAny string) ([]neutronports.Port, error) {
```

Comment:
I would use:
```
func getPorts(network *gophercloud.ServiceClient, listOpts neutronports.ListOpts) ([]neutronports.Port, error) { 
```
here, then this function can be easily used in other use cases as well. (so rows 602-608 could be in original functions)

Reply:
Make sense.